README.md

```diff
@@ -1,6 +1,199 @@
+- Run `plasmactl bump --sync` to propagate versions of updated resources to dependent resources.
+
+**Expected Outcome:**  
+No bumping or propagation occurs. However, the command should copy propagated versions from the artifact if the base version doesn’t match between the build and artifact.
+
+Test status: **OK**
+
+---
+
+#### Case 1:
+As a developer:
+- Update an existing resource (e.g., change a template, task, dependency).
+- Run `plasmactl bump` to update the resource version.
+- Run `plasmactl compose --conflicts-verbosity --skip-not-versioned`.
+- Run `plasmactl bump --sync` to propagate the new version to dependent resources.
+
+**Expected Outcome:**  
+The new version from the bump (e.g., 1111111111111) should propagate to all dependent resources as `resource_version-1111111111111`.
+
+Test status: **OK**
+
+Additional notes:
+- What if the developer didn't bump the version before propagation?
+- Should an error occur if a non-bumped or uncommitted version is set?
+
+---
+
+#### Case 2:
+As a developer:
+- Remove a resource.
+- Run `plasmactl bump`.
+- Run `plasmactl compose --conflicts-verbosity --skip-not-versioned`.
+- Run `plasmactl bump --sync` to propagate the changes.
+
+**Expected Outcome:**  
+No bumping or propagation. The developer should manually remove any dependencies before the bump.
+
+Test status: **?**
+
+---
+
+#### Case 3:
+As a developer:
+- Add a new resource.
+- Run `plasmactl bump`.
+- Run `plasmactl compose --conflicts-verbosity --skip-not-versioned`.
+- Run `plasmactl bump --sync`.
+
+**Expected Outcome:**  
+The new resource is bumped, but nothing else is propagated. Propagation occurs if the resource is overridden from another namespace (e.g., domain or package).
+
+Test status: **OK**
+
+---
+
+#### Case 4:
+Test same steps for packages; expected results should be the same.
+
+---
+
+#### Case 5:
+As a developer:
+- Update a variable in `group_vars` or `vault.yaml`.
+- Run `plasmactl bump`.
+- Run `plasmactl compose --conflicts-verbosity --skip-not-versioned`.
+- Run `plasmactl bump --sync`.
+
+**Expected Outcome:**  
+No bumping occurs, but the variable change should propagate to all dependent resources
+
+Test status: **?**
+
+---
+
+#### Case 6:
+Should this case be included?
+- Create `group_vars` or `vault.yaml` files.
+- Run `plasmactl bump`.
+- Run `plasmactl compose --conflicts-verbosity --skip-not-versioned`.
+- Run `plasmactl bump --sync`.
+
+**Expected Outcome:**  
+No bumping, but the file creation commit should propagate to dependent resources.
+
+Test status: **?**
+
+---
+
+#### Case 7:
+Should this case be included?
+- Remove `group_vars` or `vault.yaml` files.
+- Run `plasmactl bump`.
+- Run `plasmactl compose --conflicts-verbosity --skip-not-versioned`.
+- Run `plasmactl bump --sync`.
+
+**Expected Outcome:**  
+No bumping. The commit where the file was deleted should propagate to dependent resources.
```

Comment:
Variables that were deleted via file removal should propagate to all resources using them, using commit where group_var/vault was removed